vir/defs/polymorphic/ast/bodyless_method.rs

```diff
@@ -0,0 +1,60 @@
+// © 2019, ETH Zurich
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use crate::polymorphic::ast::*;
+use std::fmt;
+use std::collections::HashMap;
+
+use super::super::super::{legacy, converter};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct BodylessMethod {
+    pub name: String,
+    pub formal_args: Vec<LocalVar>,
+    pub formal_returns: Vec<LocalVar>,
+}
+
+impl fmt::Display for BodylessMethod {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "method {}(", self.name)?;
+        let mut first = true;
+        for arg in &self.formal_args {
+            if !first {
+                write!(f, ", ")?;
+            }
+            write!(f, "{:?}", arg)?;
+            first = false
+        }
+        write!(f, ") returns (")?;
+        for arg in &self.formal_returns {
+            if !first {
+                write!(f, ", ")?;
+            }
+            write!(f, "{:?}", arg)?;
+            first = false
+        }
+        write!(f, ");")
+    }
+}
+
+impl From<BodylessMethod> for legacy::BodylessMethod {
+    fn from(bodyless_method: BodylessMethod) -> legacy::BodylessMethod {
+        legacy::BodylessMethod {
+            name: bodyless_method.name,
+            formal_args: bodyless_method.formal_args.into_iter().map(|formal_arg| legacy::LocalVar::from(formal_arg)).collect(),
+            formal_returns: bodyless_method.formal_returns.into_iter().map(|formal_return| legacy::LocalVar::from(formal_return)).collect(),
+        }
+    }
+}
+
+impl converter::Generic for BodylessMethod {
+    fn substitute(self, map: &HashMap<TypeVar, Type>) -> Self {
```

Comment:
These should be not in `polymorphic`, but in `converter` module.

Reply:
Thanks. These were originally due to some structs having private fields inaccessible from different files. Now that everything is accessible, this could be achieved.